src/parsec_client/operations/psa_asymmetric_decrypt.md

```diff
@@ -27,11 +27,17 @@ Decrypt a short message with a private key. Opcode: 11 (`0x000B`)
 
 - `PsaErrorNotPermitted`: The key does not have the `decrypt` flag, or it does not permit the
    requested algorithm.
+- `PsaErrorInvalidPadding`: The decrypted padding is incorrect. See Warning below.
 
 ## Description
 
 This function will decrypt a short message with the private key of the provided key pair.
 
+**WARNING:** In some protocols, when decrypting data, it is essential that the behavior of the
+application does not depend on whether the padding is correct, down to precise timing. If the
+application must perform a decryption of unauthenticated data, the application writer must take care
+not to reveal whether the padding is invalid.
```

Comment:
The last clause makes this sentence a bit awkward to read and doesn't really add any usual information without knowing the details of the attack. You could drop it...

```suggestion
application does not depend on whether the padding is correct. If the application must perform a
decryption of unauthenticated data, the application writer must take care not to reveal whether the
padding is invalid.
```
... put it as a note in brackets
```suggestion
application does not depend on whether the padding is correct (down to precise timing). If the
application must perform a decryption of unauthenticated data, the application writer must take care
not to reveal whether the padding is invalid.
```
... or replace it with a reference (with risk of link breaking later...

```suggestion
application does not depend on whether the padding is correct (see [Bleichenbacher](ftp://ftp.rsa.com/pub/pdfs/bulletn7.pdf)). If the
application must perform a decryption of unauthenticated data, the application writer must take care
not to reveal whether the padding is invalid.
```

Reply:
@ionut-arm, Could you take a quick look at Adam's suggestions?